src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift

```diff
@@ -329,6 +326,72 @@ final class HTTPClient: Client {
 			}
 		})
 	}
+	
+	
+	func traceWarningPackageDiscovery(
+		country: String,
+		completion: @escaping TraceWarningPackageDiscoveryCompletionHandler
+	) {
+		guard let request = try? URLRequest.traceWarningPackageDiscovery(
+				configuration: configuration,
+				country: country) else {
+			completion(.failure(.requestCreationError))
+			return
+		}
+		
+		session.response(for: request, completion: { result in
+			switch result {
+			case let .success(response):
+				switch response.statusCode {
+				case 200:
+					guard let body = response.body else {
+						Log.error("Failed to unpack response body of trace warning discovery with http status code: \(String(response.statusCode))", log: .api)
+						completion(.failure(.invalidResponseError(response.statusCode)))
+						return
+					}
+					
+					do {
+						let decoder = JSONDecoder()
+						let decodedResponse = try decoder.decode(
+							TraceWarningDiscoveryResponse.self,
+							from: body
+						)
+						guard let oldest = decodedResponse.oldest,
+							  let latest = decodedResponse.latest else {
+							Log.error("Failed to get oldest or latest out of decoded response", log: .api)
+							completion(.failure(.decodingJsonError(response.statusCode)))
+							return
+						}
+						
+						let availablePackagesOnCDN = Array(oldest...latest)
+						let eTag = response.httpResponse.value(forCaseInsensitiveHeaderField: "ETag")
+						let traceWarningDiscovery = TraceWarningDiscovery(oldest: oldest, latest: latest, availablePackagesOnCDN: availablePackagesOnCDN, eTag: eTag)
```

Comment:
`availablePackagesOnCDN ` could be a computed property of TraceWarningDiscovery. No need to inject it.

Reply:
Good point. I changed it